oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/JwtEncoderAlternative.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2002-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.jwt;
+
+import java.util.function.Consumer;
+
+/**
+ * Encodes and signs JWTs, implementations may also support encryption.
+ */
+public interface JwtEncoderAlternative {
+
+	enum EncodingMode { SIGN, ENCRYPT, SIGN_THEN_ENCRYPT }
```

Comment:
I'm not aware of any use cases where applications would want to use Spring Security to only encrypt a JWT, but [this additional commit](https://github.com/jzheaux/spring-security/pull/30/commits/e7efa645d314191968ea1cf27e2aeea5d0d85308) shows that JWS, JWE, and JWS+JWE can be accommodated passively.